app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb

```diff
@@ -13,6 +13,26 @@ def raw_delete_physical_storage
     EmsRefresh.queue_refresh(self)
   end
 
+  def self.raw_validate_physical_storage(ext_management_system, options = {})
+    header_params = options[:header_params] || {}
+    header_params['Accept'] = ext_management_system.autosde_client.select_header_accept(['*/*'])
+    header_params['Content-Type'] = ext_management_system.autosde_client.select_header_content_type(['application/json'])
+
+    request_opts = {
+      :header_params => header_params,
+      :body => options.merge(
+        :system_type => PhysicalStorageFamily.find(options['physical_storage_family_id']).name,
+      ),
+      :auth_names => ['bearerAuth'],
+    }
+
+    data, status_code, headers = ext_management_system.autosde_client.call_api(:POST, 'validate-system', request_opts)
```

Comment:
Looks like an API call which should be added to the gem so you can call it without having to set up the headers/etc... yourself?

Reply:
The objects that gem uses are created by the jssonet file (generation tool) but the 'validate-credentials' wasn't implemented like that but as a static endpoint (for now). I can't use the gem function. Even though the endpoint in AutoSDE was implemented by using the generation tool, there is a function called 'storage_systems_post_with_http_info' (path: autosde_openapi_client/api/storage_system_api.rb) which can fit this API call but this function gets only StorageSystemCreate object, but I need to send different object. StorageSystemValidate for example which we didn't use the generation feature for now (as I already wrote). So, the only approach to this API call is to work with the function without using an object.

We will change the endpoint in the future to be generated by the jssonet file and then the gem can use the StorageSystemValidate object but then I'll need to create a new function as well so I can use this object instead of StorageSystemCreate.

Reply:
@agrare I've changed the implementation of the validation endpoint to be generated by the jsonnet tool (in AutoSDE). I hope we'll push it today to the master and I will use the gem function as you suggested. So the "article" that I wrote above doesn't relevant anymore.